### ISSUE TITLE
Can I use pandas dataframe in flair text classifier?

### ISSUE THREAD

**Issue by neerajnj10** [flairNLP/flair#704]
A clear and concise description of what you want to know.

Hi team,
I am new to flair and I am really impressed by the documentation but I am having tough time in understanding the implementation of flair with my dataset in particular. 

I have a dataframe df with two columns that looks like this:
**TEXT  |  Label

04/04/2017-gggg | 5
Email yyyy | 1
1/13/2016 -xxxxx | 5
senetive data | 5
so filling with rubbish | 5**

My question is I have 5 classes that I want to predict using the text data in Text column but I do not see any example following pandas dataframe as an input, can we not do it ? DO I have to store the data as txt file, and labels in separate txt file?
I am confused little bit here, can you please provide any example for my case that I could get started with for text classifier.

Also make_label_dictionary does not work for dataframe columns. what am I missing?

**Comment by ortizbje** [comment]
Hello, 
This should help you : https://towardsdatascience.com/text-classification-with-state-of-the-art-nlp-library-flair-b541d7add21f

**Comment by stefan-it** [comment]
@neerajnj10 See also this issue #491 -> there's a nice code snippet that shows how to use pandas to convert data frames into a flair compatible format :)

**Comment by neerajnj10** [comment]
Thank you so much for help @stefan-it and @ortizbje, really appreciate it. I was able to make it run for my project with it. 
I have one small issue though, I want to do class weighting as well, since my data is "highly imbalanced" and results are not good. Any suggestions please? Are there any implementations for imbalanced data?

**Comment by alanakbik** [comment]
Hello @neerajnj10 I've been looking into imbalanced data (see #678) but so far progress has been slow. I've looked into adapting the loss function to produce higher loss for rare classes, but it takes a lot of experiment-specific fine-tuning and also at least in my experiments sacrifices precision for recall. Now I'm looking into weighted data sampling during training, i.e. upsampling rare classes to make sure the model trains for them. So work is ongoing but I cannot say when we make more progress on this.
